### PR TITLE
Dzint/component multi mesh from tag

### PR DESCRIPTION
**Pull request by daniel-zint**
This component adds a child mesh that is a manifold representation from tags that were potentially non-manifold.

### CODE REVIEW

File: components/wmtk_components/multimesh_from_tag/wmtk/components/multimesh_from_tag/multimesh_from_tag.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        keeps.emplace_back(substructure_label);
+        mesh.clear_attributes(keeps);
+    }
+
+    {
+        std::map<std::string, std::vector<int64_t>> names;
+        names[options.input] = mesh.absolute_multi_mesh_id();
+        names[options.output] = mmft.substructure()->absolute_multi_mesh_id();
+        names[options.output + "_soup"] =
+            mmft.substructure_soup()->absolute_multi_mesh_id(); // TODOfix: should be deleted
```
Comment (mtao) [inline components/wmtk_components/multimesh_from_tag/wmtk/components/multimesh_from_tag/multimesh_from_tag.cpp:51]: remember to delete

File: src/wmtk/Tuple.hpp [hunk 2]
```diff
@@ -78,6 +78,10 @@ class Tuple
     /// Checks if a tuple is "null". This merely implies the global index is -1
     bool is_null() const;
     Tuple with_updated_hash(int64_t new_hash) const;
+
+    int8_t local_vid() const;
```
Comment (mtao) [inline src/wmtk/Tuple.hpp:82]: IIRC we purposely were leaving these as implementation details - please don't make public

Comment (mtao) [reply src/wmtk/Tuple.hpp:82]: (just to increase the friction of most people using these to do things, like access topological attributes directly)

File: src/wmtk/attribute/AttributeManager.hpp [hunk 3]
```diff
@@ -86,6 +86,9 @@ class AttributeManager : public wmtk::utils::MerkleTreeInteriorNode
 
     std::string get_name(const attribute::TypedAttributeHandleVariant& attr) const;
 
+    template <typename T>
+    void set_name(const TypedAttributeHandle<T>& attr, const std::string& name);
```
Comment (mtao) [inline src/wmtk/attribute/AttributeManager.hpp:90]: just to be clear-  this should only be public in the manager class but the mesh class doesn't expose this right?

File: src/wmtk/simplex/tuples_preserving_primitive_types.hpp [hunk 4]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include <vector>
+#include <wmtk/Tuple.hpp>
+
+
+namespace wmtk::simplex {
+
+/**
+ * @brief Compute all tuples that contain simplex(ptype1, t) and simplex(ptype2, t).
```
Comment (mtao) [inline src/wmtk/simplex/tuples_preserving_primitive_types.hpp:10]: and are contained by simplex(ptype2) ?

File: src/wmtk/simplex/tuples_preserving_primitive_types.cpp [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+    Tuple t_iter = t;
+
+    std::vector<Tuple> intersection_tuples;
+    do {
+        intersection_tuples.emplace_back(t_iter);
+        for (size_t i = 1; i < switch_tuple_types.size() - 1; ++i) {
+            t_iter = mesh.switch_tuple(t_iter, switch_tuple_types[i]);
+        }
+    } while (t != t_iter);
```
Comment (mtao) [inline src/wmtk/simplex/tuples_preserving_primitive_types.cpp:26]: does this general function really work via circulation? i think we might ahve discussed how this is implemented in some nice way, but i don't undertand it right now

**Review by mtao — COMMENTED** [review]
some things to change / discuss perhaps but generally fine

File: components/wmtk_components/multimesh_from_tag/wmtk/components/multimesh_from_tag/internal/MultiMeshFromTag.hpp [hunk 6]
```diff
@@ -0,0 +1,113 @@
+#pragma once
+
+#include <deque>
+#include <wmtk/TetMesh.hpp>
+#include <wmtk/TriMesh.hpp>
+#include <wmtk/io/HDF5Writer.hpp>
```
[the remark this answers is not in the document]
Comment (daniel-zint) [reply components/wmtk_components/multimesh_from_tag/wmtk/components/multimesh_from_tag/internal/MultiMeshFromTag.hpp:6]: Seem like I forgot to clean them up. I removed them now and also a bunch of other unnecessary includes.

File: src/wmtk/attribute/AttributeManager.hpp (hunk 3 above)
Comment (daniel-zint) [reply src/wmtk/attribute/AttributeManager.hpp:90]: Correct. It is not in the mesh class at all.

File: src/wmtk/simplex/tuples_preserving_primitive_types.cpp (hunk 5 above)
Comment (daniel-zint) [reply src/wmtk/simplex/tuples_preserving_primitive_types.cpp:26]: I hope it does. It worked in my few test cases and I cannot think of a case where it does not work but I cannot guarantee that it works all the time.

**Review by mtao — APPROVED** [review]
had forgtten one other comment apparently

File: src/wmtk/Tuple.hpp (hunk 2 above)
Comment (daniel-zint) [reply src/wmtk/Tuple.hpp:89]: I made the component a friend of Tuple and made the getters private.

**Pull request merged**